### PR TITLE
Kudzu Rebalance

### PR DESCRIPTION
**Pull request by Crazylemon64**
:cl:Crazylemon
fix: Kudzu should now be less of (but still) a station-destroying force of nature.
tweak: Bluespace kudzu now only pierces a single layer of walls before losing its "charge".
tweak: Aggressive spread now does not bust rwalls, but does attack obstructing objects in its path.
tweak: Kudzu now only spreads on space turfs when it has specific mutations to do so.
tweak: Kudzu events should be more interesting on their own.
/:cl:

### CODE REVIEW

**Comment by ktoma36** [comment]
I do agree that it should not eat through reinforced walls and such, that is silly.

File: code/modules/events/spacevine.dm [hunk 1]
```diff
@@ -500,10 +519,18 @@
 	processing_objects.Add(src)
 	init_subtypes(/datum/spacevine_mutation/, mutations_list)
 	if(potency != null)
-		mutativeness = potency / 10
+		// 1 mutativeness at 10 potency
+		// 4 mutativeness at 100 potency
+		mutativeness = log(10, potency)*log(10, potency)
```
Comment (Crazylemon64) [inline code/modules/events/spacevine.dm]: shoot 0 potency is a thing 1 sec

**Comment by Fox-McCloud** [comment]
I'm highly skeptical of making walls more resistant to explosions *purely* for the sake of Kudzu.

If explosive Kudzu is the problem, it would make more sense to nerf that, than to buff something that is generally otherwise fine for everything else.

**Comment by Crazylemon64** [comment]
There's no weaker form of destruction these vines can exert, sans doofy special-casing - either they can destroy both regular walls and rwalls, or they can't scratch either.

**Comment by Crazylemon64** [comment]
There, kudzu now will leave rwalls alone without touching other code

**Comment by SigholtStarsong** [comment]
Does this address the absofreakingly absurd speed at which Kudzu spreads?

**Comment by Crazylemon64** [comment]
@SigholtStarsong The seed->vine scaling has been adjusted to be within more reasonable bounds

**Comment by Kyep** [comment]
Kudzu needs to be nerfed, and hard.
I'm neutral on the exact method - but a major nerf is definitely required.

**Comment by Crazylemon64** [comment]
@Kyep This **is** a major nerf

Bluespace now only penetrates to 1 room over, instead of making it a station-devouring omnivine

**Comment by ktoma36** [comment]
Oh that round was me messing with the genes of the Kudzu, regular kudzu is not nearly as bad

**Pull request merged**